### ISSUE TITLE
Add filter options for global_search in json settings

### ISSUE THREAD

**Issue by bhbmaster** [zed-industries/zed#19333]
### Check for existing issues

- [X] Completed

### Describe the feature

I often have to search a subset of my entire directory structure using global search using filters. My work flow is such that I open my work directory as my root directory for Zed and other code/text editors and work from there.
I have multiple directories in my work directory but why I search I only need to look thru `_notes` and `_cases`.

Example:
root directory is `~/work`
Which has the following dirs and each with thousands of files:
```
_notes
_cases
_bin
docx
HR
etc
```

I only want to be able to search thru _notes and _cases. I can do this with the UI global search filter option. However, if I close out of the global search those settings are lost.

Into include I put:
`_notes/*, _cases/*`

Side-Question: is the syntax that I have for the include filter correct to allow searching recursively thru _notes and _cases? There are dozens to hundreds of sub-dirs inside _notes and _cases.

The only editable options I found for the search / global search is this:

```
{
  "search": {
    "whole_word": false,
    "case_sensitive": false,
    "include_ignored": false,
    "regex": true
  }
}
```

Here is a screenshot of my settings mid search. The results ended up coming out:

<img width="1031" alt="Zed 2024-10-16 17 58 52" src="https://github.com/user-attachments/assets/954d90a7-0424-49d9-b394-2a604b78bae6">

Note: i checked and currently this appears to be the first request for this feature. There are mentions of updating file_scan_exclusions or .gitignore. However those have additional affects that I would not want to affect my search. For example I want to be able to still see the other directories. I just want global search to always be bound by my include and exclude parameters if I am working in my work directory.

### If applicable, add mockups / screenshots to help present your vision of the feature

Just add the "filter" json key into the "search" json key & make it understood by the editor.

```
{
  "search": {
    "whole_word": false,
    "case_sensitive": false,
    "include_ignored": false,
    "regex": true
    "filters": {
        "include": "_notes/*, _cases/*",
        "exclude": "" 
      }
  }
}
```

In my case I would put these in `~/work/.zed/settings.json`. However, if someone wants settings for any zed env then they should be able to put it in `~/.config/zed/settings.json`

**Comment by notpeter** [comment]
> `_notes/*, _cases/*`
> Side-Question: is the syntax that I have for the include filter correct to allow searching recursively thru _notes and _cases? There are dozens to hundreds of sub-dirs inside _notes and _cases.

Yes. This should work. You can also use more explicit glob matching (`_notes/**/*.md, **/_cases/**/*`) depending on how your folders are structured.

**Comment by gunnartorfis** [comment]
This would be amazing in for example monorepos. 

In my case I want to include .ts & .tsx files by default but exclude .go files. We'd also want to exclude .next, .turbo, .output, .vinxi, etc and also some OpenAPI spec generation code.

**Comment by miranyx** [comment]
Can I take this on?

**Comment by klener89** [comment]
I currently use Pycharm on a very large monorepo and the search scopes are incredibly useful.
Especially switching between the different scopes makes it very valuable.
E.g. I can define a scope for `tests only` or `no_tests` and switch between them, depending in what part of the code I want to search.

When implementing the possibility to define and persist filters, I would love if they could be dynamically changed. Defining them by project, as proposed here, seems reasonable.

I attached a screenshot from Pycharm to show the select field where I can dynamically define the search filter:
<img width="1217" height="890" alt="Image" src="https://github.com/user-attachments/assets/5f4932ec-9c36-48e2-8983-3a817c1ec0fa" />

**Comment by slavnycoder** [comment]
Yes, please!!!

**Comment by Marius-Muis** [comment]
Bump, would like this as well.

**Comment by amritk** [comment]
Yep would help when switching over!

**Comment by jackTabsCode** [comment]
This would be incredibly useful for us: our game contains tons of serialized "scene" files containing data structures that also exist in code. It happens very frequently that I'll search for something and get a couple results in our source files and thousands of results in large scene files.  However, these scene files are still useful and I don't want to entirely remove them from Zed's index. The ability to kinda just automatically ignore them and then back those ignores out when I want, would be great.
